### ISSUE TITLE
Purge does not work with Azurite

### ISSUE THREAD

**Issue by michiel-sunweb** [Azure/azure-functions-durable-extension#2406]
### Description

Using the purge command (`func durable purge-history` or `HTTP DELETE http://localhost:7071/runtime/webhooks/durabletask/instances/:instanceId`) results in an error.

### Expected behavior

Using the purge command with a valid instance ID should not result in an error, and should delete both the instance's record in the Instances table, as well as the related records in the History table.

### Actual behavior
Only the instance's record from the Instances table is deleted, and the following error is shown:

> Element 0 in the batch returned an unexpected response code.

When using the purge command via HTTP, then the 500 response includes a stack trace:

```
{
    "Message": "Something went wrong while processing your request",
    "ExceptionMessage": "Element 0 in the batch returned an unexpected response code.",
    "ExceptionType": "DurableTask.AzureStorage.Storage.DurableTaskStorageException",
    "StackTrace": "   at DurableTask.AzureStorage.Storage.AzureStorageClient.MakeStorageRequest[T](Func`3 storageRequest, String accountName, String operationName, String clientRequestId, Boolean force) in /_/src/DurableTask.AzureStorage/Storage/AzureStorageClient.cs:line 140\n   at DurableTask.AzureStorage.Storage.Table.ExecuteBatchAsync(TableBatchOperation batchOperation, String batchType) in /_/src/DurableTask.AzureStorage/Storage/Table.cs:line 172\n   at DurableTask.AzureStorage.Storage.Table.ExecuteBatchAsync(IList`1 entityBatch, String batchType, Func`3 batchOperation) in /_/src/DurableTask.AzureStorage/Storage/Table.cs:line 150\n   at DurableTask.AzureStorage.Storage.Table.DeleteBatchAsync(IList`1 entityBatch) in /_/src/DurableTask.AzureStorage/Storage/Table.cs:line 122\n   at DurableTask.AzureStorage.Tracking.AzureTableTrackingStore.<>c__DisplayClass49_0.<<DeleteAllDataForOrchestrationInstance>b__1>d.MoveNext() in /_/src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs:line 725\n--- End of stack trace from previous location ---\n   at DurableTask.AzureStorage.Tracking.AzureTableTrackingStore.DeleteAllDataForOrchestrationInstance(OrchestrationInstanceStatus orchestrationInstanceStatus) in /_/src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs:line 737\n   at DurableTask.AzureStorage.Tracking.AzureTableTrackingStore.PurgeInstanceHistoryAsync(String instanceId) in /_/src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs:line 768\n   at Microsoft.Azure.WebJobs.Extensions.DurableTask.AzureStorageDurabilityProvider.PurgeInstanceHistoryByInstanceId(String instanceId) in D:\\a\\_work\\1\\s\\src\\WebJobs.Extensions.DurableTask\\AzureStorageDurabilityProvider.cs:line 132\n   at Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableClient.Microsoft.Azure.WebJobs.Extensions.DurableTask.IDurableOrchestrationClient.PurgeInstanceHistoryAsync(String instanceId) in D:\\a\\_work\\1\\s\\src\\WebJobs.Extensions.DurableTask\\ContextImplementations\\DurableClient.cs:line 563\n   at Microsoft.Azure.WebJobs.Extensions.DurableTask.HttpApiHandler.HandleDeleteHistoryByIdRequestAsync(HttpRequestMessage request, String instanceId) in D:\\a\\_work\\1\\s\\src\\WebJobs.Extensions.DurableTask\\HttpApiHandler.cs:line 565\n   at Microsoft.Azure.WebJobs.Extensions.DurableTask.HttpApiHandler.HandleRequestAsync(HttpRequestMessage request) in D:\\a\\_work\\1\\s\\src\\WebJobs.Extensions.DurableTask\\HttpApiHandler.cs:line 380"
}
```

### Relevant source code snippets

Unknown.

### Known workarounds

After running the purge command, I manually delete the records from the History table.

### App Details

- **Durable Functions extension version**: 2.9.0
- **Azure Functions runtime version**: v4
- **Programming language used**: C#

> If you don't want to share your Function App or storage account name GitHub, please at least share the orchestration instance ID. Otherwise it's extremely difficult to look up information.

**Comment by davidmrdavid** [comment]
Hi @michiel-sunweb, thanks for reaching out.
I took a look at this and tried using the HTTP API and I got an expected response of 

```JSON
{
    "instancesDeleted": 1
}
```

However, my HTTP DELETE request also included my taskhub name and local azurite storage authorization code. If I remove both of these parameters, I get a 404 error, not a 500. Can you please confirm that `azurite` is running in the background? And can you please try using the HTTP URL provided by `CreateCheckStatusResponse` under `statusQueryGetUri` to perform your `DELETE` request? That should include the taskhub and auth-code.

As an aside - can you please provide me with an example of how you're running the delete request using `func`? A fully parametrized example would help me debug from that end as well. Thanks!

**Comment by michiel-sunweb** [comment]
Hey @davidmrdavid,

> Can you please confirm that azurite is running in the background? 

I can confirm. I have also used Microsoft Azure Storage Explorer to verify that the `InitialLoadPushHistory` and `InitialLoadPushInstances` tables are populated as I run/debug orchestrations.

> And can you please try using the HTTP URL provided by `CreateCheckStatusResponse` under `statusQueryGetUri` to perform your `DELETE` request? That should include the taskhub and auth-code.

This part of your response is a bit difficult to understand for me. I will first briefly explain how I think the HTTP API works, but then I think it will be easier to just focus on the command-line scenario, so we can avoid any misunderstanding.

First, looking at the [documentation](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-http-api) I see that some HTTP endpoints return an object with a `statusQueryGetUri` property. But also a `purgeHistoryDeleteUri` property. Surely the `purgeHistoryDeleteUri` is the relevant one?

Second, you mention that you included the taskhub name, but according to the documentation this is optional, and if not specified, the "the current function app's task hub name is assumed." That seems a better approach, as I don't want to purge the history of a different task hub.

And third, you mention an azurite storage account code, but I do not think the HTTP API defines such a thing. It does specify a `connection` query string parameter, but that is defined as "The name of the connection app setting for the backend storage provider" and that, if not provided, "the default connection configuration for the function app is assumed." While I am not sure what you mean by "azurite storage account code", I think here a value like `AzureWebJobsStorage` would be valid.

Fourth, you mention an auth code that would eb included in the `statusQueryGetUri` value returned by the HTTP API. According to the documentation, a `code` query string parameter may be included in such URLs, but that is defined as " authorization key required to invoke the API." Since I am running locally, there is no such key in play here.

So with that out of the way, I think it's much easier going forward to focus on the command-line scenario below.

> can you please provide me with an example of how you're running the delete request using `func`?

I open a terminal in the directory of the Azure Function app project. This directory contains the `.csproj` file, the `local.settings.json` file and the `host.json` file. `local.settings.json` contains `"AzureWebJobsStorage": "UseDevelopmentStorage=true"` under `"Values"`. `host.json` contains `"hubName": "InitialLoadPush"` under `"extensions": "durableTask"`.

From the terminal I can run `func durable get-instances` without any additional options to get a list of instances formatted as JSON. For this my Azure Function app does not even need to be running (although it does need to be built it seems). Then I just run `func durable purge-history` (no additional option) and get the error.

tl;dr: `func durable purge-history`

**Comment by michiel-sunweb** [comment]
Just for extra emphasis: I am running the Azure Function app **locally** against a local instance of Azurite.

**Comment by davidmrdavid** [comment]
Thanks for the detailed answer, @michiel-sunweb. I agree it would make things easier to focus on the CLI `func durable` command, so let's focus on that **for now**. It's worth noting that the core Durable Functions team does not directly own the CLI so this might require engaging the responsible team.

As of today, I'm getting some issues with Azurite in general (errors at app startup) , so I'll need to try  to reproduce this with a fresh installation. I'll get to that as soon as possible.
